sdk/python/kfp/client/client.py

```diff
@@ -1360,6 +1360,129 @@ def get_run(self, run_id: str) -> kfp_server_api.V2beta1Run:
         """
         return self._run_api.run_service_get_run(run_id=run_id)
 
+    def get_logs(
+        self,
+        run_id: str,
+        component_name: Optional[str] = None,
+        namespace: Optional[str] = None,
+    ) -> Union[str, Dict[str, str]]:
+        """Downloads logs from a pipeline run's components.
+
+        Args:
+            run_id: ID of the pipeline run.
+            component_name: Optional name of specific component to get logs from.
+                           If None, returns logs from all components.
+            namespace: Kubernetes namespace to use. Used for multi-user deployments.
+                      For single-user deployments, this should be left as None.
+
+        Returns:
+            If component_name is specified: string containing the component logs.
+            If component_name is None: dictionary mapping component names to their logs.
+
+        Raises:
+            ValueError: If the run_id doesn't exist or component_name is not found.
+            ApiException: If there's an error accessing Kubernetes API.
+
+        Example:
+          ::
+
+            client = kfp.Client()
+            # Get logs from specific component
+            logs = client.get_logs(
+                run_id='5d08dd3d-58d9-4d02-9ff2-f2cee8dbfda8',
+                component_name='foo2'
+            )
+            print(logs)
+
+            # Get logs from all components
+            all_logs = client.get_logs(
+                run_id='5d08dd3d-58d9-4d02-9ff2-f2cee8dbfda8'
+            )
+            for component, log_text in all_logs.items():
+                print(f"Component {component}: {log_text[:100]}...")
+        """
+        try:
+            from kubernetes import client as k8s_client
+            from kubernetes import config as k8s_config
+            from kubernetes.client.rest import ApiException
+        except ImportError:
+            raise ImportError('Kubernetes package is not installed. '
+                              'Install it using: pip install kubernetes')
+
+        namespace = namespace or self.get_user_namespace()
+
+        run = self.get_run(run_id=run_id)
```

Comment:
I'd rather this leverage the log artifacts automatically uploaded by the KFP launcher. This has the benefit of still working even if the pod gets deleted and it doesn't require the user to have access to the Kubernetes cluster itself.